instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go

```diff
@@ -33,24 +36,90 @@ type gRPCContextKey struct{}
 type gRPCContext struct {
 	messagesReceived int64
 	messagesSent     int64
+	metricAttrs      []attribute.KeyValue
+}
+
+type handler struct {
+	tracer             trace.Tracer
+	meter              metric.Meter
+	rpcDuration        metric.Float64Histogram
+	rpcRequestSize     metric.Int64Histogram
+	rpcResponseSize    metric.Int64Histogram
+	rpcRequestsPerRPC  metric.Int64Histogram
+	rpcResponsesPerRPC metric.Int64Histogram
+}
```

Comment:
Why do we even need this handler?

Can we put all the instruments, providers, meter and tracer in `config` and initialize everything in  `newConfig`?

Reply:
Yes, we can. 

To be honest, I also consider putting every into config at first. But, then I found there already exists a metric `rpcServerDuration` which is already used in the implementation of `interceptor` and conflicts with the name `rpcDuration` we use for `stats.Handler`.  If we put them together, I thought maybe confused for users. So I add a new `handler` struct.

So how about we put all those five metric `rpcDuration`... into config, and keep the original `rpcServerDuration`, just like following

```
type config struct {
	Filter           Filter
	Propagators      propagation.TextMapPropagator
	TracerProvider   trace.TracerProvider
	MeterProvider    metric.MeterProvider
	SpanStartOptions []trace.SpanStartOption

	ReceivedEvent bool
	SentEvent     bool

	meter             metric.Meter
	rpcServerDuration metric.Int64Histogram

	rpcDuration        metric.Float64Histogram
	rpcRequestSize     metric.Int64Histogram
	rpcResponseSize    metric.Int64Histogram
	rpcRequestsPerRPC  metric.Int64Histogram
	rpcResponsesPerRPC metric.Int64Histogram
}
```

When the implementation of interceptor is deprecated, the we remove the old `rpcServerDuration` metric?  WDYT?
@pellared

Reply:
How about updating the `interceptor` to use the new `rpcDuration`?

Reply:
> How about updating the `interceptor` to use the new `rpcDuration`?

That works for me, I will try to work like that

Reply:
Done